Graphics issues in PPSSPP 1.19.1 (Appimage)

### Game or games this happens in

all

### What area of the game

After new game

### What happens

I have been using PPSSPP for many years and it has always worked well, but after the release of version 1.19.1 I went to the project's github and downloaded the appimage "PPSSPP-v1.19.1-anylinux-x86_64.AppImage" and started having several problems.

First, the emulator never goes full screen, even when changing the option in the settings:

![Image](https://github.com/user-attachments/assets/f8c32af0-fafe-41d5-b17b-97ff09a20fe6)

Second, the graphics backend cannot be changed in the settings, it is turned off and in Vulkan:

![Image](https://github.com/user-attachments/assets/f5e9c409-6ffd-4348-96f2-5ddd9fb2fc32)

Sometimes the emulator gives an error when opening and then closes, other times it opens with the problems I mentioned earlier, I'll leave the terminal log here when it gives an error:

.local/share/applications/PPSSPP-v1.19.1-anylinux-x86_64.AppImage
11:40:947 Vulkan/VulkanLoader.cpp:462 I[G3D]: VulkanMayBeAvailable: Device allowed ('SDL:Linux')
11:40:949 Vulkan/VulkanLoader.cpp:401 I[G3D]: Vulkan library loaded ('libvulkan.so')
11:40:949 Vulkan/VulkanLoader.cpp:511 I[G3D]: VulkanMayBeAvailable: Enumerating instance extensions
11:40:979 Vulkan/VulkanLoader.cpp:522 I[G3D]: VulkanMayBeAvailable: Instance extension count: 24
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_device_group_creation (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_display (00000017)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_external_fence_capabilities (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_external_memory_capabilities (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_external_semaphore_capabilities (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_get_display_properties2 (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_get_physical_device_properties2 (00000002)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_get_surface_capabilities2 (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_surface (00000019)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_surface_protected_capabilities (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_wayland_surface (00000006)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_xcb_surface (00000006)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_xlib_surface (00000006)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_drm_display (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_xlib_display (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (0000000a)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000002)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_direct_mode_display (00000001)
11:40:983 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_display_surface_counter (00000001)
11:40:984 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_headless_surface (00000001)
11:40:984 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_surface_maintenance1 (00000001)
11:40:984 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_swapchain_colorspace (00000005)
11:40:984 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_portability_enumeration (00000001)
11:40:984 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_LUNARG_direct_driver_loading (00000001)
11:40:984 Vulkan/VulkanLoader.cpp:568 I[G3D]: VulkanMayBeAvailable: Calling vkCreateInstance
11:40:993 Vulkan/VulkanLoader.cpp:575 I[G3D]: VulkanMayBeAvailable: Vulkan test instance created successfully.
MESA-INTEL: warning: Ivy Bridge Vulkan support is incomplete
11:40:997 Vulkan/VulkanLoader.cpp:617 I[G3D]: VulkanMayBeAvailable: Eligible device found: 'Intel(R) HD Graphics 4000 (IVB GT2)'
11:40:997 Vulkan/VulkanLoader.cpp:634 I[G3D]: VulkanMayBeAvailable: Found working Vulkan API!
11:40:997 Vulkan/VulkanLoader.cpp:641 I[G3D]: VulkanMayBeAvailable: Destroying instance
DEBUG: Vulkan might be available.
[W][00:11:41.042780] pw.conf      | [          conf.c: 1182 try_load_conf()] can't load config client.conf: No such file or directory
[E][00:11:41.043219] pw.conf      | [          conf.c: 1215 pw_conf_load_conf_for_context()] can't load config client.conf: No such file or directory
Info: We compiled against SDL version 2.32.56 and we are linking against SDL version 2.32.56. :)
11:41:329 Thread/ThreadManager.cpp:228 I[SYSTEM]: ThreadManager::Init(compute threads: 4, all: 8)
11:41:329 Util/PortManager.cpp:475 I[SCENET]: UPnPService: Begin of UPnPService Thread
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/local/share/ppsspp/assets
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /tmp/.mount_PPSSPremp13009023090147297219/bin/assets
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /tmp/.mount_PPSSPremp13009023090147297219/bin
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/local/share/ppsspp/assets
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/local/share/games/ppsspp/assets
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/share/ppsspp/assets
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/share/games/ppsspp/assets
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : assets
11:41:330 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /home/dirceu
11:41:332 Core/Config.cpp:403 I[G3D]: Longest display side: 1366 pixels. Choosing scale 2
SDL_CreateWindow failed for GL 4.6: Invalid window
SDL_CreateWindow failed for GL 4.5: Invalid window
SDL_CreateWindow failed for GL 4.4: Invalid window
SDL_CreateWindow failed for GL 4.3: Invalid window
SDL_CreateWindow failed for GL 4.2: Invalid window
SDL_CreateWindow failed for GL 4.1: Invalid window
SDL_CreateWindow failed for GL 4.0: Invalid window
SDL_CreateWindow failed for GL 3.3: Invalid window
SDL_CreateWindow failed for GL 3.2: Invalid window
SDL_CreateWindow failed for GL 3.1: Invalid window
SDL_CreateWindow failed for GL 3.0: Invalid window
SDL_CreateWindow failed: Invalid window
GL init error '' - falling back to Vulkan
MESA-INTEL: warning: Ivy Bridge Vulkan support is incomplete
(/__w/ppsspp/ppsspp/UI/NativeApp.cpp:NativeInitGraphics:810:10000000): [g_screenManager] (menu, 0.5s) No screenmanager, bad init order. Backend = 3

It seems to be something with beckend, it is not possible to switch to opengl and vulkan is not rendering in full screen:

![Image](https://github.com/user-attachments/assets/73430c0b-7cc2-4709-afe1-9e8ef7b9c4b6)

Here is the log of when the emulator opened without crashing and ran a game and then I exited the emulator normally:

 .local/share/applications/PPSSPP-v1.19.1-anylinux-x86_64.AppImage
18:14:889 Vulkan/VulkanLoader.cpp:462 I[G3D]: VulkanMayBeAvailable: Device allowed ('SDL:Linux')
18:14:890 Vulkan/VulkanLoader.cpp:401 I[G3D]: Vulkan library loaded ('libvulkan.so')
18:14:890 Vulkan/VulkanLoader.cpp:511 I[G3D]: VulkanMayBeAvailable: Enumerating instance extensions
18:14:919 Vulkan/VulkanLoader.cpp:522 I[G3D]: VulkanMayBeAvailable: Instance extension count: 24
18:14:921 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_device_group_creation (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_display (00000017)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_external_fence_capabilities (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_external_memory_capabilities (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_external_semaphore_capabilities (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_get_display_properties2 (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_get_physical_device_properties2 (00000002)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_get_surface_capabilities2 (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_surface (00000019)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_surface_protected_capabilities (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_wayland_surface (00000006)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_xcb_surface (00000006)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_xlib_surface (00000006)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_drm_display (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_acquire_xlib_display (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_report (0000000a)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_debug_utils (00000002)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_direct_mode_display (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_display_surface_counter (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_headless_surface (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_surface_maintenance1 (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_EXT_swapchain_colorspace (00000005)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_KHR_portability_enumeration (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:530 I[G3D]: VulkanMaybeAvailable: Instance extension found: VK_LUNARG_direct_driver_loading (00000001)
18:14:922 Vulkan/VulkanLoader.cpp:568 I[G3D]: VulkanMayBeAvailable: Calling vkCreateInstance
18:14:931 Vulkan/VulkanLoader.cpp:575 I[G3D]: VulkanMayBeAvailable: Vulkan test instance created successfully.
MESA-INTEL: warning: Ivy Bridge Vulkan support is incomplete
18:14:935 Vulkan/VulkanLoader.cpp:617 I[G3D]: VulkanMayBeAvailable: Eligible device found: 'Intel(R) HD Graphics 4000 (IVB GT2)'
18:14:935 Vulkan/VulkanLoader.cpp:634 I[G3D]: VulkanMayBeAvailable: Found working Vulkan API!
18:14:935 Vulkan/VulkanLoader.cpp:641 I[G3D]: VulkanMayBeAvailable: Destroying instance
DEBUG: Vulkan might be available.
[W][00:18:14.978677] pw.conf      | [          conf.c: 1182 try_load_conf()] can't load config client.conf: No such file or directory
[E][00:18:14.979058] pw.conf      | [          conf.c: 1215 pw_conf_load_conf_for_context()] can't load config client.conf: No such file or directory
Info: We compiled against SDL version 2.32.56 and we are linking against SDL version 2.32.56. :)
18:15:172 Thread/ThreadManager.cpp:228 I[SYSTEM]: ThreadManager::Init(compute threads: 4, all: 8)
18:15:172 Util/PortManager.cpp:475 I[SCENET]: UPnPService: Begin of UPnPService Thread
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/local/share/ppsspp/assets
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /tmp/.mount_PPSSPremp13009023090147297219/bin/assets
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /tmp/.mount_PPSSPremp13009023090147297219/bin
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/local/share/ppsspp/assets
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/local/share/games/ppsspp/assets
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/share/ppsspp/assets
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /usr/share/games/ppsspp/assets
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : assets
18:15:172 VFS/VFS.cpp:14 D[IO]: Registered VFS for prefix : /home/dirceu
18:15:173 Core/Config.cpp:403 I[G3D]: Longest display side: 1366 pixels. Choosing scale 2
18:15:176 UI/NativeApp.cpp:308 E[LOADER]: Failed graphics backends: OPENGL
18:15:176 Core/Config.cpp:519 E[LOADER]: Graphics backend failed for 0, trying another
MESA-INTEL: warning: Ivy Bridge Vulkan support is incomplete
18:24:418 Core/System.cpp:593 N[BOOT]: PPSSPP v1.19-56-g435b265888
18:24:809 root         N[G3D]: Vulkan/ShaderManagerVulkan.cpp:617 ShaderCache: Loaded 6 vertex, 13 fragment shaders and 0 geometry shaders (failed 0)
18:24:809 root         N[G3D]: Vulkan/PipelineManagerVulkan.cpp:785 Creating 37 pipelines from cache (1x MSAA)...
18:24:844 root         N[BOOT]: UI/EmuScreen.cpp:381 Booted /media/dirceu/66760EC6760E96C9/Games/Roms/psp/Ultimate Ghosts'n Goblins (USA).iso...
18:37:829 TitleExec    E[ME]: HLE/sceMpeg.cpp:1688 0=sceMpegQueryPcmEsSize(09ed3e58, 09f978c0, 09f978c4): UNIMPL
18:37:832 TitleExec    E[SCEKERNEL]: HLE/sceKernelThread.cpp:2646 NOT_SUSPEND=sceKernelResumeThread(352): sceKernelResumeThread(352): thread not suspended
18:37:832 MovieOpenTas E[SCEKERNEL]: HLE/sceKernelThread.cpp:2612 ILLEGAL_THID=sceKernelSuspendThread(352): cannot suspend current thread
18:37:840 MovieOpenTas E[SCEKERNEL]: HLE/sceKernelThread.cpp:2612 ILLEGAL_THID=sceKernelSuspendThread(352): cannot suspend current thread
18:42:253 ScePsmfPlaye E[SCENET]: HLE/sceNet.cpp:951 0=sceNetFreeThreadinfo(331): UNIMPL
18:48:992 TitleExec    E[ME]: HLE/sceMpeg.cpp:1688 0=sceMpegQueryPcmEsSize(09ed3e58, 09f97820, 09f97824): UNIMPL
18:48:992 TitleExec    E[SCEKERNEL]: HLE/sceKernelThread.cpp:2646 NOT_SUSPEND=sceKernelResumeThread(382): sceKernelResumeThread(382): thread not suspended
18:48:992 MovieOpenTas E[SCEKERNEL]: HLE/sceKernelThread.cpp:2612 ILLEGAL_THID=sceKernelSuspendThread(382): cannot suspend current thread
18:48:993 MovieOpenTas E[SCEKERNEL]: HLE/sceKernelThread.cpp:2612 ILLEGAL_THID=sceKernelSuspendThread(382): cannot suspend current thread
18:53:881 ScePsmfPlaye E[SCENET]: HLE/sceNet.cpp:951 0=sceNetFreeThreadinfo(361): UNIMPL
18:53:949 PlayLoad     E[SCEKERNEL]: HLE/sceKernelThread.cpp:2612 ILLEGAL_THID=sceKernelSuspendThread(392): cannot suspend current thread
18:54:336 AreaProcess  E[SCEKERNEL]: HLE/sceKernelThread.cpp:2004 NOT_DORMANT=sceKernelStartThread(293, 4, 09f92960)
18:54:337 AreaLoad     E[SCEKERNEL]: HLE/sceKernelThread.cpp:2612 ILLEGAL_THID=sceKernelSuspendThread(407): cannot suspend current thread
18:54:668 AreaInit     E[SCEKERNEL]: HLE/sceKernelThread.cpp:2612 ILLEGAL_THID=sceKernelSuspendThread(408): cannot suspend current thread
19:33:060 Vulkan/ShaderManagerVulkan.cpp:645 N[G3D]: Saved 6 vertex and 13 fragment shaders
19:33:060 Vulkan/PipelineManagerVulkan.cpp:723 N[G3D]: Saved Vulkan pipeline ID cache (37 unique pipelines/37).
Leaving main



### What should happen

In the previous version available (1.18.1 flatpak) I have no problem, I can switch between opengl and vulkan normally, and the games and the emulator render in full screen:

![Image](https://github.com/user-attachments/assets/fa46167b-1dfd-4b86-89a7-c346d5c7f139)

![Image](https://github.com/user-attachments/assets/487af9aa-abb3-4b63-92d4-a8aa4b561474)

![Image](https://github.com/user-attachments/assets/543dcbd4-388d-400f-8b1d-0aba48325b8e)

### GE frame capture

_No response_

### Platform

Linux / BSD

### Mobile device model or graphics card (GPU)

Intel Hd Graphics 4000 

### PPSSPP version affected

1.19-56-g435b26

### Last working version

1.18.1

### Graphics backend (3D API)

Vulkan

### Checklist

- [x] Test in the [latest git build](https://www.ppsspp.org/devbuilds) in case it's already fixed.
- [x] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [ ] Try changing graphics settings to determine if one causes the glitch (especially speed hacks or enhancements/replacements.)
- [ ] Include logs or screenshots of issue.

Regarding sdl2 deps, when i tried to installed it from source at `https://github.com/libsdl-org/SDL/archive/refs/tags/release-2.32.8.tar.gz` it failed during "Create AppImage" step on aarch64, because PPSSPP could no longer detects x11:
```
Failed to initialize SDL with joystick support. Retrying without.
Unable to initialize SDL: x11 not available
DEBUG: Vulkan is not available, not using Vulkan.
```
While it have no problem on the x86_64 one, i haven't figured out which SDL2 dependency packages causing this issue.

Anyway, i couldn't make ppsspp to use the sdl2 libs and headers, because the default `make install` is installed at /usr/local and PPSSPP keeps using the one from archlinux at /usr, and trying to change the path using `./configure --prefix=/usr` keeps getting "configure: error: C compiler cannot create executables" error.

I will try installing it from AUR like you said, hopefully the additional deps won't breaks the aarch64 job like the deps for Arch at https://github.com/libsdl-org/SDL/blob/main/docs/README-linux.md

Hmm.. the SDL2 from AUR seems to be missing the CMakeLists.txt
```
:: Proceed with installation? [Y/n] 
checking keyring...
checking package integrity...
loading package files...
checking for file conflicts...
:: Processing package changes...
installing sdl2...
Optional dependencies for sdl2
    alsa-lib: ALSA audio driver [installed]
    libpulse: PulseAudio audio driver [installed]
    jack: JACK audio driver [installed]
    pipewire: PipeWire audio driver [installed]
    libdecor: Wayland client decorations [installed]
installing sdl2-debug...
:: Running post-transaction hooks...
(1/3) Arming ConditionNeedsUpdate...
(2/3) Cleaning up package cache...
(3/3) Cleaning up package cache...
/__w/ppsspp/ppsspp/sdl2/build /__w/ppsspp/ppsspp/sdl2
CMake Warning:
  Ignoring extra path from command line:

   ".."


CMake Error: The source directory "/__w/ppsspp/ppsspp/sdl2" does not appear to contain CMakeLists.txt.
```

I'm not sure from which line is this error:
```
Cloning into 'sdl2'...
==> ERROR: sdl2 is not available for the 'aarch64' architecture.
==> ERROR: arch does not allow empty values.
Error: Process completed with exit code 12.
```

when installing it with:
```
# Install SDL2 from AUR
sed -i 's|EUID == 0|EUID == 69|g' /usr/bin/makepkg # because the docker image is ran as root this is needed lol
git clone https://aur.archlinux.org/sdl2.git
cd sdl2
sed -i "s|x86_64|$ARCH|" ./PKGBUILD
makepkg -f
pacman --noconfirm -Rsndd sdl2-compat
pacman --noconfirm -U *.pkg.tar.*
# End of Install SDL2 from AUR
```

--- a/.github/workflows/appimage.yml
+++ b/.github/workflows/appimage.yml
@@ -94,6 +94,7 @@ jobs:
           sed -i 's/DownloadUser/#DownloadUser/g' /etc/pacman.conf
 
           ARCH=$(uname -m)
+          echo "ARCH=$ARCH" >> $GITHUB_ENV # Export ARCH to be reused on other steps
           if [[ "$ARCH" == "x86_64" ]]; then
           	PKG_TYPE='x86_64.pkg.tar.zst'
           else
@@ -112,6 +113,7 @@ jobs:
             desktop-file-utils \
             git \
             cmake \
+            ninja \
             clang \
             ccache \
             llvm-libs \
@@ -125,10 +127,10 @@ jobs:
             libc++ \
             libc++abi \
             glew \
-            sdl2 \
             sdl2_ttf \
             python \
             patchelf \
+            pipewire \
             pipewire-audio \
             pulseaudio \
             pulseaudio-alsa \
@@ -137,8 +139,12 @@ jobs:
             xorg-server-xvfb \
             zsync \
             vulkan-nouveau \
-            vulkan-radeon 
-
+            vulkan-radeon \
+            vulkan-headers \
+            alsa-lib sndio hidapi ibus jack libdecor libgl libpulse libusb \
+            libx11 libxcursor libxext libxinerama libxkbcommon libxrandr libxrender libxss libxtst \
+            wayland wayland-protocols
+          
           if [[ "$ARCH" == "x86_64" ]]; then
             pacman -Syu --noconfirm vulkan-intel
           fi
@@ -179,6 +185,17 @@ jobs:
 
           #clang --version
           #clang++ --version
+
+          # Install SDL2 from AUR
+          sed -i 's|EUID == 0|EUID == 69|g' /usr/bin/makepkg # because the docker image is ran as root this is needed lol
+          git clone https://aur.archlinux.org/sdl2.git
+          pushd sdl2
+          sed -i "s|x86_64|$ARCH|" ./PKGBUILD
+          makepkg -f
+          pacman --noconfirm -Rsndd sdl2-compat
+          pacman --noconfirm -U *.pkg.tar.*
+          popd
+          # End of Install SDL2 from AUR
 
           # Build ppsspp
           #./b.sh --release --no-sdl2 --no-png

